FordLabs/PeopleMover
# Spike to calculate end date of assignment based off of algorithm similar to the one used to calculate start date

## Issue
NA (No issue entered for this yet. Maybe part of #250 (Performance issue related to Assignments needing end dates)? 

## What was done
- Backend: 
  - Add endDate to Assignment as a transient variable
  - Add method to AssignmentRepository to get assignments after specified date
  - Implement findEndDate using future assignment dates
  - Update calculateStartDate method to calculateStartAndEndDates
- Frontend:
  - Add endDate to Assignment as an optional undefined variable
  - Add endDate to AssignmentHistory class (in PersonForm)
  - Update assignment history component method to support end dates
- Add/update unit tests

## How to test
1. Open PeopleMover
2. Go into a space with a bunch of assignments and people movement, or create such a space
3. Look at open person for editing and look at their assignment history.
4. If they are on an assignment that has an end date, it should have 'Assigned to <product> from <startDate> to <endDate>
5. If they are on an assignment that does not have an end date (is still ongoing), it should have 'Moved to <product> on <startDate> (already existing functionality)
